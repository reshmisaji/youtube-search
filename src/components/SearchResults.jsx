import InfiniteScroll from "react-infinite-scroll-component"
import '../styles/SearchResults.css'
import { Suspense } from "react";

const VideoCard = (data, index) => {
  const { snippet, id } = data;

  return <div data-testid={`video-card-${index}`} key={`video-card-${index}`} className="video-card-container" >
    <Suspense fallback={
      <img alt={snippet?.description} src={snippet?.thumbnails?.high?.url} className="thumbnail" />
    }>
      <iframe
        src={`https://www.youtube.com/embed/${id.videoId}`}
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture;"
        allowFullScreen
        className="thumbnail"></iframe>
    </Suspense>
    <div className="details">
      <div className="title">{snippet?.title}</div>
      <div className="channel-logo"></div>
      <div className="channel-name">{snippet?.channelTitle}</div>
      <div className="views"></div>
      <div className="likes"></div>
      <button className="play-video"></button>
      <button className="like"></button>
      <button className="dislike"></button>
    </div>
  </div>

}

const SearchResults = ({ results, fetchData, hasMore }) => {
  return (<div data-testid="search-results" id="search-results" className="results-container">
    <InfiniteScroll
      className="results"
      dataLength={results.length}
      next={fetchData}
      hasMore={hasMore}
      loader={<h4>Loading...</h4>}
      endMessage={
        <p style={{ textAlign: 'center' }}>
          <b>Yay! You have seen it all</b>
        </p>
      }>
      {results?.map(VideoCard)}</InfiniteScroll>
  </div>)
}
export default SearchResults