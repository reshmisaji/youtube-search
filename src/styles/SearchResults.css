.results-container {
    padding-top: 70px;
}

.results {
    text-align: center;
    margin: 0px 10vw 0px 10vw;
}

.video-card-container {
    display: flex;
    margin: 20px 0px 20px 0px;
}

.thumbnail {
    border-radius: 12px;
    cursor: pointer;
    box-shadow: 1px 1px 10px 1px #58575359;
    min-width: 480px;
    min-height: 360px;
    background: black;
}

.details {
    font-family: system-ui;
    text-align: left;
    margin-left: 20px;
}

.title {
    font-weight: bold;
    font-size: 23px;
    font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
}

.channel-name,
.title,
.description {
    word-break: break-word;
}

.channel-logo {
    border-radius: 50%;
    min-width: 35px;
    min-height: 35px;
    background: #3d3b3b;
    color: white;
    font-size: 25px;
    font-weight: 500;
    display: flex;
    justify-content: center;
    align-items: center;
}

.channel {
    display: flex;
    align-items: center;
    padding-top: 15px;
}

.channel-name {
    padding-left: 10px;
    font-size: 18px;
    font-weight: bold;
    font-family: monospace;
    color: #3d3b3b;
}

.publish-date {
    font-size: 12px;
    font-weight: bold;
    color: #333131;
}

.description {
    padding-top: 15px;
    font-size: 16px;
    font-family: system-ui;
    font-weight: 500;
    color: #3d3b3b;
}

a{
    text-decoration: none;
}

a:focus-visible{
    outline: none;
}

@media screen and (max-width: 1024px) {
    .results {
        margin: 0px 5vw 0px 5vw;
    }
}

@media screen and (max-width:800px) {
    .results {
        margin: 0px 10vw 0px 10vw;
    }

    .video-card-container {
        flex-direction: column;
    }

    .publish-date,
    .description {
        display: none;
    }

    .details {
        margin: 0;
    }

    .title {
        font-weight: 500;
        font-size: 20px;
    }

    .channel {
        padding-top: 0px;
    }

    .channel-logo {
        min-width: 17px;
        min-height: 17px;
        font-size: 15px;
        background: #6f6b6b;
    }

    .channel-name {
        padding-left: 5px;
        font-size: 18px;
        color: #6f6b6b;
    }

    .video-card-container {
        margin: 15px 0px 35px 0px;
    }
}

@media screen and (max-width:605px) {
    .thumbnail {
        min-width: 95vw;
        min-height: 267px;
    }

    .results {
        margin: 0px 2vw 0px 2vw;
    }
}