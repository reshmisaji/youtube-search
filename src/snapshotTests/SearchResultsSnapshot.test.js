import React from "react";
import renderer from "react-test-renderer";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
    it("Should render search resuls with props", () => {
        const fetchResults = jest.fn()
        const searchResult = []
        const actualDOM = renderer.create(<SearchResults results={searchResult} fetchData={fetchResults} hasMore={true} />).toJSON()
        expect(actualDOM).toMatchSnapshot()
    })

    it("Should render search resuls with props", () => {
        const fetchResults = jest.fn()
        const searchResult = [{
            "kind": "youtube#searchResult",
            "etag": "BTEODBcIlxK7VY72sAxSV06I_y4",
            "id": {
                "kind": "youtube#video",
                "videoId": "Y9faB5tXij0"
            },
            "snippet": {
                "publishedAt": "2021-11-19T14:00:10Z",
                "channelId": "UCH5d2bpWBVu6eQk1amUSpyA",
                "title": "Johnny Stimson - Flower (Piano Version)",
                "description": "Follow Johnny on Instagram: http://bit.ly/2iqacB1",
                "thumbnails": {
                    "default": {
                        "url": "https://i.ytimg.com/vi/Y9faB5tXij0/default.jpg",
                        "width": 120,
                        "height": 90
                    },
                    "medium": {
                        "url": "https://i.ytimg.com/vi/Y9faB5tXij0/mqdefault.jpg",
                        "width": 320,
                        "height": 180
                    },
                    "high": {
                        "url": "https://i.ytimg.com/vi/Y9faB5tXij0/hqdefault.jpg",
                        "width": 480,
                        "height": 360
                    }
                },
                "channelTitle": "Johnny Stimson",
                "liveBroadcastContent": "none",
                "publishTime": "2021-11-19T14:00:10Z"
            }
        }]
        const actualDOM = renderer.create(<SearchResults results={searchResult} fetchData={fetchResults} hasMore={true} />).toJSON()
        expect(actualDOM).toMatchSnapshot()
    })
})