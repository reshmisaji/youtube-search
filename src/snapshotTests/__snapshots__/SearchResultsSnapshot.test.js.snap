// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults Should render search resuls with props 1`] = `
<div
  className="results-container"
  data-testid="search-results"
  id="search-results"
>
  <div
    className="infinite-scroll-component__outerdiv"
    style={Object {}}
  >
    <div
      className="infinite-scroll-component results"
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "height": "auto",
          "overflow": "auto",
        }
      }
    >
      <h4>
        Loading...
      </h4>
    </div>
  </div>
</div>
`;

exports[`SearchResults Should render search resuls with props 2`] = `
<div
  className="results-container"
  data-testid="search-results"
  id="search-results"
>
  <div
    className="infinite-scroll-component__outerdiv"
    style={Object {}}
  >
    <div
      className="infinite-scroll-component results"
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "height": "auto",
          "overflow": "auto",
        }
      }
    >
      <div
        className="video-card-container"
        data-testid="video-card-0"
      >
        <iframe
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture;"
          allowFullScreen={true}
          className="thumbnail"
          src="https://www.youtube.com/embed/Y9faB5tXij0"
        />
        <div
          className="details"
        >
          <div
            className="title"
          >
            Johnny Stimson - Flower (Piano Version)
          </div>
          <div
            className="channel-logo"
          />
          <div
            className="channel-name"
          >
            Johnny Stimson
          </div>
          <div
            className="views"
          />
          <div
            className="likes"
          />
          <button
            className="play-video"
          />
          <button
            className="like"
          />
          <button
            className="dislike"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
